rosbag2_compression/src/rosbag2_compression/zstd_compressor.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+
+#include "logging.hpp"
+#include "rosbag2_compression/zstd_compressor.hpp"
+
+namespace rosbag2_compression
+{
+
+std::string ZstdCompressor::compress_uri(const std::string & uri)
+{
+  auto start = std::chrono::high_resolution_clock::now();
+  auto compressed_uri = uri + "." + get_compression_identifier();
+  std::ifstream infile(uri);
+  if (infile.is_open()) {
```

Comment:
It looks like `is_open` doesn't guarantee that there were no errors in the stream.
https://stackoverflow.com/questions/24097580/ifstreamis-open-vs-ifstreamfail

Reply:
Interesting... did not know that!

Reply:
I actually didn't know that until I googled it. I did know that you are supposed to check the stream after opening and before reading.